### DISCUSSION TITLE
Get All Cell Precedents For A Cell Range

### DISCUSSION THREAD

**Discussion by BrianHung** [handsontable/hyperformula#1352] [API Questions]
I want to get the list of all cell precedents when `setCellContents` is called.

When `value: RawCellContent` is a single value and `setCellContents(address, value)` is called, `getCellPrecedents(address)` works to get all the precedents in the new formula.

However, what happens when `values: RawCellContent[][]` is an array? Is there an alternative besides brute-forcing `getCellPrecedents` for all the addresses that are affected by `setCellContents(address, values)`?

The API for `getCellPrecedents` says it can take a `SimpleCellRange`

```
HyperFormula.getCellPrecedents(address: SimpleCellAddress | SimpleCellRange)

If the argument is a range of cells, getCellPrecedents() returns some of the cell addresses and smaller ranges contained in that range (but not all of them). The exact result depends on the optimizations applied by the HyperFormula to the dependency graph, some of which are described in the section ["Optimizations for large ranges"](vscode-file://vscode-app/Applications/Cursor.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html).
```

but it looks like it only works when the exact address is used as a precedent, not for smaller ranges contained in that range (as mentioned in the API comment).

**Comment by adrianszymanski89** [comment]
Hi @BrianHung 

Thank you for your message. I think it would be best if @sequba would recommend something here.

**Comment by sequba** [comment]
@BrianHung 

> The API for `getCellPrecedents` says it can take a `SimpleCellRange` [...] but it looks like it only works when the exact address is used as a precedent, not for smaller ranges contained in that range (as mentioned in the API comment).

Can you give an example (demo) of a situation when `getCellPrecedents` does not work as expected?
